src/Lean/Meta/Basic.lean

```diff
@@ -1560,7 +1580,42 @@ def lambdaTelescope (e : Expr) (k : Array Expr → Expr → n α) (cleanupAnnota
   If `cleanupAnnotations` is `true`, we apply `Expr.cleanupAnnotations` to each type in the telescope.
 -/
 def lambdaBoundedTelescope (e : Expr) (maxFVars : Nat) (k : Array Expr → Expr → n α) (cleanupAnnotations := false) : n α :=
-  map2MetaM (fun k => lambdaTelescopeImp e false (.some maxFVars) k (cleanupAnnotations := cleanupAnnotations)) k
+  map2MetaM (fun k => lambdaTelescopeImp e true false true false (.some maxFVars) k (cleanupAnnotations := cleanupAnnotations)) k
+
+/--
+Given `e` of the form `let x₁ := v₁; ...; let xₙ := vₙ; A`, executes `k xs A`,
+where `xs` is an array of free variables for the binders.
+The `let`s can also be `have`s.
+
+- If `cleanupAnnotations` is `true`, applies `Expr.cleanupAnnotations` to each type in the telescope.
+- If `preserveNondep` is `false`, all `have`s are converted to `let`s.
+- If `nondepLetOnly` is `true`, then only `have`s are consumed (it stops at the first dependent `let`).
+
+See also `mapLetTelescope` for entering and rebuilding the telescope.
+-/
+def letTelescope (e : Expr) (k : Array Expr → Expr → n α) (cleanupAnnotations := false) (preserveNondepLet := true) (nondepLetOnly := false) : n α :=
```

Comment:
I am assuming there is no way to consume only dependent `let`s.

Reply:
Correct, there's no way to do that at the moment (I haven't needed it yet).